.changes/set_native_icon_signature.md

```diff
@@ -0,0 +1,5 @@
+---
+"tauri": 'patch:enhance'
+---
+
+remove mut self from set_native_icon signature
```

Comment:
```suggestion
---
"tauri": 'patch:bug'
---

Changed `IconMenuItem::set_native_icon` signature to take `&self` instead of `&mut self` to fix compilation error on macos.
```